### PR TITLE
Redesign worker parameter contracts

### LINKED ISSUE #996: `parse_parameters()` does not match the connector parameter schema.
`parse_parameters()` does not match the connector parameter schema. The schema supplies `SSL_CHECK`, `PROXY_SERVER`, `ADDITIONAL_HEADERS`, `USER_AGENT`, and `USE_GLOBAL_PROXY`, while the worker reads `SSL`, `PROXIES`, and `HEADERS`. As a result, SSL verification and proxy/header settings are silently ignored. 

Please normalize these using the same parsing behavior as `MispCollector`, and parse `REQUEST_TIMEOUT` from its stored string with a safe default. Add a test using the real connector fixture, whose timeout is currently an empty string.

_edited scope_

_Originally posted by @armins in https://github.com/taranis-ai/taranis-ai/pull/984#pullrequestreview-4905437464_

### PR DESCRIPTION
**Pull request by b3n4kh**
## Summary

Replace database-backed worker definitions and per-value validation metadata with a shared Pydantic registry in `src/models`. Each supported worker type now has one authoritative parameter contract for field names, native types, defaults, validation, ordering, labels, tooltip descriptions, secret handling, and exceptional UI hints.

Configured resources persist only explicit uppercase parameter names and canonical string values in JSON. Core validates configurations before persistence and expands defaults for execution, while workers validate the effective payload again before dispatch.

Fixes #996 by aligning MISP connector parsing with the registered `SSL_CHECK`, `PROXY_SERVER`, and `ADDITIONAL_HEADERS` parameters.

## What changed

- Add a complete worker-definition and parameter-model registry with `extra="forbid"`, native Pydantic types, JSON Schema form metadata, and shared canonical string conversion.
- Generate frontend parameter forms from validation JSON Schema, including titles, tooltip descriptions, constraints, defaults, secret widgets, cron/template/word-list hints, and runtime choices.
- Add JSON `parameters` storage and shared create/replace/merge/remove/configured/effective serialization for sources, bots, connectors, product types, and publisher presets.
- Add parameter-aware `PATCH`; keep `PUT` as replacement, preserve omitted secrets, support `null` removal, and reject worker-type changes.
- Mask secrets in normal APIs and add permission-checked, audited, non-cacheable explicit reveal endpoints.
- Validate active configurations before commit and enqueueing; disabled incomplete sources and bots remain editable but cannot be enabled or executed.
- Normalize and validate source imports atomically, and validate pre-seeded configured overrides through the registry.
- Remove worker catalog pre-seeding, legacy worker/parameter ORM models and admin APIs, rule-to-pattern form logic, and the Worker Type admin UI.
- Remove unsupported email/twitter collector remnants and stale HTML presenter parameters; add frontend/core PPN collector filtering.
- Add the `1.5.0` changelog entry and worker-parameter agent memory.

## Migration and deployment

The yoyo migration aggregates legacy owner parameter rows into canonical JSON, accepts equal duplicates, aborts on conflicting duplicates, removes unknown keys, drops invalid optional values, and rejects active configurations that remain invalid or incomplete. It then removes the legacy worker, parameter-value, enum, and join tables.

This is a coordinated core/frontend/worker/models release. Deployment documentation requires a verified database snapshot before migration. Rollback is restoring that snapshot and redeploying the previous images; there is intentionally no reconstructive downgrade or compatibility-table path.

## Update and secret semantics

- `POST` validates and stores explicitly submitted values, including values equal to defaults.
- `PUT.parameters` replaces the complete non-secret configuration; omitted secrets are preserved.
- `PATCH.parameters` merges keys; `null` removes an optional configured value; omitted keys are preserved.
- Normal list/detail responses expose configured non-secret values and secret markers only.
- Worker-authenticated responses receive validated effective values with defaults expanded and raw secrets included.

## Validation

- Ruff: models, core, frontend, and worker passed.
- Core: `879 passed`.
- Frontend unit: `332 passed, 53 skipped`; djlint passed.
- Worker: `195 passed`.
- Targeted frontend user E2E: `10 passed`; focused admin/RQ regression scenarios passed.
- The complete frontend E2E rerun encountered broad fixture/browser startup errors before assertions; the previously failing user file passed cleanly in isolation and no product-code failure was reproducible.
- Migration verified as a yoyo migration against a PostgreSQL database upgraded from release `1.4.3`, with focused migration coverage for conversion, duplicate handling, validation, canonicalization, and legacy-table removal.

## Security note

Secrets remain unencrypted at rest, matching current project behavior. This change prevents routine API and page exposure; encryption and key management remain separate work.

### CODE REVIEW

File: src/worker/worker/collectors/collector_tasks.py [hunk 1]
```diff
@@ -117,6 +118,7 @@ def collector_task(osint_source_id: str, manual: bool = False):
 
     try:
         source = collector.get_source(osint_source_id)
+        source["parameters"] = effective_parameter_values(source["type"], source.get("parameters", {}))
```
[the remark this answers is not in the document]
Comment (b3n4kh) [reply src/worker/worker/collectors/collector_tasks.py:121]: This task-result gap is already covered by the registered RQ failure bridge: `start_worker()` installs `rq_failure_exception_handler`, and `get_task_name_from_job()` resolves collector, collector-preview, and presenter task identities. Uncaught parameter-validation exceptions therefore persist a terminal `FAILURE`; the bridge tests cover generic presenter and collector failures plus function-name fallback.

File: .github/copilot-instructions.md [hunk 2]
```diff
@@ -8,3 +8,4 @@
 - Respect the frontend and API boundary rules in `AGENTS.md`.
 - Do not introduce `pip`-based setup or ad hoc dependency installation.
 - Keep changes small, reversible, and documented when behavior or setup changes.
+- During PR reviews, add a reviewer briefing comment with the most important pointers and hints (such as risky areas, key files, and useful verification steps) and, most importantly, a clear estimate of how long a human review should take (for example, `Estimated human review time: 20–30 minutes`).
```
Comment (not4Pedro) [inline .github/copilot-instructions.md:11]: 👍

File: src/core/core/managers/queue_manager.py
Comment (not4Pedro) [inline src/core/core/managers/queue_manager.py:1]: Are in this file for sure not leaking `exc` stack traces?

**Review by not4Pedro — APPROVED** [review]

File: src/core/core/managers/queue_manager.py
Comment (b3n4kh) [reply src/core/core/managers/queue_manager.py:1]: fixed
